src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1948,8 +1948,8 @@ public override BoundNode VisitNullCoalescingAssignmentOperator(BoundNullCoalesc
 
             // The assignment to the left below needs the declared type from VisitLvalue, but the hidden
             // unnecessary check diagnostic needs the current adjusted type of the slot
-            VisitLvalue(leftOperand);
-            TypeSymbolWithAnnotations targetType = _resultType;
+            Visit(leftOperand);
```

Comment:
>Visit [](start = 12, length = 5)

:memo: I'm a bit unsure whether we should use `VisitRValue` instead (which does `Visit` then `Unsplit`).